lib/posix/clock.c

```diff
@@ -126,3 +128,36 @@ int gettimeofday(struct timeval *tv, void *tz)
 
 	return res;
 }
+
+clock_t times(struct tms *buffer)
+{
+	clock_t rtime;
+	clock_t utime;
+	clock_t stime;
+	struct timespec ts;
+
+	BUILD_ASSERT(CLOCKS_PER_SEC == USEC_PER_SEC, "POSIX requires CLOCKS_PER_SEC to be 1000000");
+
+	if (clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &ts) < 0) {
+		return -1;
+	}
+
+	rtime = k_ticks_to_us_floor64(k_uptime_ticks());
+	utime = ts.tv_sec * USEC_PER_SEC + ts.tv_nsec / NSEC_PER_USEC;
+	stime = rtime - utime;
```

Comment:
Seems like `stime` here is the time not spent in the process?  I..e, time spent in other processes.  That seems like a bad choice to return.  For instance, any code that wants to find the percentage of CPU time used by the current process would add utime and stime together than then divide by real time.  This would result in all processes appearing to use 100% CPU.

Reply:
See previous comment